src/v/datalake/arrow_schema_translator.cc

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "datalake/arrow_schema_translator.h"
+
+#include "iceberg/datatypes.h"
+
+#include <arrow/type.h>
+#include <arrow/type_fwd.h>
+#include <arrow/type_traits.h>
+
+#include <memory>
+#include <stdexcept>
+
+namespace datalake {
+class schema_converter_interface {
+public:
+    virtual ~schema_converter_interface() noexcept = default;
+    // Return an Arrow field descriptor for this converter.
+    virtual std::shared_ptr<arrow::Field>
+    field(const std::string& name, iceberg::field_required required) = 0;
```

Comment:
should this have a virtual destructor?

Reply:
I believe the virtual destructor is required for this.